### PR TITLE
full spectrum driver

### PR DESCRIPTION
**Pull request by volvonk**
Adds support to full spectrum laser cutters, supports only cutting, uses a fixed speed interpolator, which is good enough for low speeds, there are some things to improve(see TODO list in FullSpectrumCutter.java) but at least the protocol and building of packets are already implemented.

### CODE REVIEW

File: src/com/t_oster/liblasercut/drivers/FullSpectrumCutterProperty.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  {
+    if ("power".equals(name))
+    {
+      return (Float) 100f;
+    }
+    else if ("speed".equals(name))
+    {
+      return (Float) 100f;
+    }
+    else if ("focus".equals(name))
```
Comment (t-oster) [inline src/com/t_oster/liblasercut/drivers/FullSpectrumCutterProperty.java]: this also?

File: src/com/t_oster/liblasercut/drivers/FullSpectrumCutterProperty.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  {
+    if ("power".equals(name))
+    {
+      return (Float) 0f;
+    }
+    else if ("speed".equals(name))
+    {
+      return (Float) 0f;
+    }
+    else if ("focus".equals(name))
```
Comment (t-oster) [inline src/com/t_oster/liblasercut/drivers/FullSpectrumCutterProperty.java]: is this a leftover from copy-paste?

File: src/com/t_oster/liblasercut/drivers/FullSpectrumCutterProperty.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+/**
+ * Handles parameters for full spectrum jobs.
+ * 
+ * Based upon PowerSpeedFocusProperty by Thomas Oster <thomas.oster@rwth-aachen.de>
+ * 
+ * @author Volkan Vonk <vol.vonk@yandex.com> 
+ */
+public class FullSpectrumCutterProperty implements LaserProperty
+{
```
Comment (t-oster) [inline src/com/t_oster/liblasercut/drivers/FullSpectrumCutterProperty.java]: This class could be used on other devices and should thus be named `FloatPowerSpeedProperty` in order to be consistent with the other properties.

File: src/com/t_oster/liblasercut/drivers/FullSpectrumCutterProperty.java (hunk 2 above)
Comment (volvonk) [inline src/com/t_oster/liblasercut/drivers/FullSpectrumCutterProperty.java]: yeah, sorry about that

File: src/com/t_oster/liblasercut/drivers/FullSpectrumCutterProperty.java (hunk 1 above)
Comment (volvonk) [inline src/com/t_oster/liblasercut/drivers/FullSpectrumCutterProperty.java]: yeah, sorry again

**Comment by volvonk** [comment]
Hello, thank you, i made the requested changes
The driver it is not complete, because it lacks of a acc/dec stage so for high speeds 60< the inertia begins to degrade the job quality, though very long and complex jobs may be performed with great accuracy at lower speeds. I think would be better to use lower speed settings by default maybe 600 or 500 steps/sec, what do you think about this?
Also the driver doesnt support engraving, but i may add some insights about engraving packets.

**Pull request merged**